### ISSUE TITLE
[Feature request] Add gradient clip 

### ISSUE THREAD

**Issue by classicsong** [awslabs/graphstorm#289]
In some cases, we may encounter exploding gradient problem where large loss gradients accumulate during back propagation, which will eventually result in very large weight updates during training. Using gradient clipping can avoid certain problems. ([torch.nn.utils.clip_grad_value_])

**Comment by zheng-da** [comment]
can you provide more detailed explanation of this feature request? in what cases, it is required?
